app/models/transformation_mapping/vm_migration_validator.rb

```diff
@@ -37,7 +37,7 @@ def identify_vms
     conflict_list = []
 
     vm_names = @vm_list.collect { |row| row['name'] }
-    vm_objects = Vm.where(:name => vm_names, :ems_cluster => mapped_clusters).includes(:lans, :storages, :host, :ext_management_system)
+    vm_objects = Vm.where(:name => vm_names)
```

Comment:
For better performance, we want to do the SQL query once, then look at the in-memory VM objects for selection.
Should we have here:
```
 vm_objects = Vm.where(:name => vm_names).includes(:ems_cluster, :lans, :storages, :host, :ext_management_system)
```

Then later on line 55, we do:
```
vms = vm_objects.select { |vm| mapped_clusters.include?(vm.ems_cluster) }
```

Reply:
Thanks @lfu - made the above change